rust/schema/annotations.py

```diff
@@ -197,9 +204,10 @@ class _:
     foo(1) = 4;
     ```
     """
+    arg_list: drop
```

Comment:
Is this really necessary? I think it would be useful to have fields in a subclass that overlap with the base class as a means to override their types with a more specific one.

Reply:
the generation machinery would choke on that, as fields in `schema.py` (unless marked with synth) all translate to tables in the DB scheme, so we would get problems there.

Consider that the customizable hand-written QL classes can indeed override the getters to change the return types, though admittedly that might get messy with some derived predicates (like `getAnX`).

`synth` fields might be close to that use case, though there might be a problem with missing `override`. If we do need to cater for this use case, we can probably concoct something up.